dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -516,6 +609,66 @@ def _perform_t_test(
         results["welch"]["p-value"] = float(welch_p_val)
         return results
 
+    @staticmethod
+    def _calculate_psi(
+        self_match_count: int,
+        histogram1: np.ndarray,
+        other_match_count: int,
+        histogram2: np.ndarray,
+    ) -> Optional[float]:
+        """
+        Calculate PSI (Population Stability Index).
+
+        ```
+        PSI = SUM((other_pcnt - self_pcnt) * ln(other_pcnt / self_pcnt))
+        ```
+
+        PSI Breakpoint Thresholds:
+            - PSI < 0.1: no significant population change
+            - 0.1 < PSI < 0.2: moderate population change
+            - PSI >= 0.2: significant population change
+
+        :param self_match_count: self.match_count
+        :type self_match_count: int
+        :param histogram1: self._stored_histogram["histogram"]
+        :type histogram1: np.ndarray
+        :param self_match_count: other_profile.match_count
+        :type self_match_count: int
+        :param histogram2: other_profile._stored_histogram["histogram"]
+        :type histogram2: np.ndarray
+        :return: psi_value
+        :rtype: optional[float]
+        """
+        psi_value = 0
+        invalid_stats = False
+
+        if isinstance(histogram2["bin_edges"], type(None)) or isinstance(
+            histogram1["bin_edges"], type(None)
+        ):
+            warnings.warn(
```

Comment:
I think we use logger.warn in the repo? So that a user can shut off warnings? I cannot recall.

Reply:
I was basing this off `_perform_t_test`, but lemme double check

Reply:
You are correct! My mistake, thanks!